stats/player.py

```diff
@@ -34,44 +23,20 @@ def get_id_set(self):
             s.add(player.player_id)
         return s
 
-    def add_player(self, season_id):
+    def generate_rows(self, season_id):
         """
         Build GET REST request to the NBA for a season.
-        Since we're in the context of a single player, we'll assemble
-        rows one at a time then do a bulk insert at the end.
         """
         params = self.build_params(season_id)
 
         # Encode without safe '+', apparently the NBA likes unsafe url params.
         params_str = urllib.parse.urlencode(params, safe=':+')
-
-        # json response
-        response = (
-            requests
-            .get(url=self.player_info_url, headers=headers, params=params_str)
-            .json()
-        )
-
-        # pulling just the data we want
-        player_info = response['resultSets'][0]['rowSet']
-
-        # looping over data to insert into table
-        # direct array is a PITA, how can we abstract this...
-        for row in player_info:
-            new_row = {
-                'player_id': row[0],
-                'player_name': row[1],
-                'college': row[8],
-                'country': row[9],
-                'draft_year': row[10],
-                'draft_round': row[11],
-                'draft_number': row[12]
-            }
-            self.rows.append(new_row)
+        super().generate_rows(season_id, params_str)
```

Comment:
`self.`. No reason to call super here.

Reply:
I don't fully understand. If `generate_rows` is defined in the current class, calling `self.generate_rows` will resolve the base class' method? Python's OO is a bit magic to me, especially only having experience in Java to Erlang. Just confirming: `self.generate_rows` will then call the base class' `generate_rows` function?

Reply:
Ooops, my bad, I didn't realize you were calling this inside `generate_rows`. I was a bit confused with the highlighting. I am more used to Bitbucket, and I just realized I can get a split view in Github too. 

No, this is correct. I will mention that, in general, Python supports multi-inheritance, and this is why `super` was introduced. If it's single inheritance, you can refer to the parent class directly, i.e., `GenericRequester.generate_rows()`. 

That's preferred over using super in single inheritance because there is no need for resolution. You don't need Python to figure out who the parent is.

Reply:
Cool thanks for clarifying. I'll make that change as part of the next PR

Reply:
It's not a big deal.

Unrelated to this, I had been thinking about the draft last week, and I wanted to get some answers......except, I didn't know how. My SQL skills are very basic.

I was wondering if you would be okay if I reached out to you in private to discuss such queries. Your email address is in the git logs.

Or I can just raise an issue here.

Reply:
Feel free to shoot me an email.